sphinxcontrib/images.py

```diff
@@ -265,13 +265,15 @@ def download_images(app, env):
                       .format(src, dst))
 
 
-def configure_backend(app):
-    global DEFAULT_CONFIG
+def check_config(app, config):
+    '''Ensure all config values are defined'''
 
-    config = copy.deepcopy(DEFAULT_CONFIG)
-    config.update(app.config.images_config)
-    app.config.images_config = config
+    merged = copy.deepcopy(DEFAULT_CONFIG)
+    merged.update(config.images_config)
+    config.images_config = merged
```

Comment:
So what is the rationale behind changing `config` as opposed to `app.config`?

In `configure_backend(app)` we get the `images_config` dictionary from `config = app.config.images_config`, so the goal of `check_config(app, config)` must be to update `app.config`, otherwise the updates would not be available in `configure_backend(app)` which only takes `app` as parameter...

Reply:
I assume this is just inconsistency in Sphinx but `app.config` _is_ `config`, but since `config` is passed as a parameter here I figured it would make sense to use the parameter, otherwise it's unused.